model-optimizer/extensions/middle/UnsqueezeTileReshapeBlockToInterpolate.py

```diff
@@ -76,79 +80,110 @@ def pattern(self):
 
         d_idx = int(second_input_of_unsqueeze.value)
 
         second_input_of_tile = match['tile'].in_port(1).get_connection().get_source().node
         if not second_input_of_tile.has_valid('value'):
-            return
+            return False
 
         input_shape_of_unsqueeze = unsqueeze_node.in_port(0).data.get_shape()
-        if len(input_shape_of_unsqueeze) not in {4, 5}:
+        input_rank_of_unsqueeze = len(input_shape_of_unsqueeze)
+        if input_rank_of_unsqueeze not in {4, 5}:
+            return False
+
+        if input_rank_of_unsqueeze + 1 != len(second_input_of_tile.value):
+            return False
+
+        expected_tile_constant = np.ones(input_rank_of_unsqueeze + 1)
+        expected_tile_constant[d_idx] = float(second_input_of_tile.value[d_idx])
+
+        if not np.array_equal(expected_tile_constant, float32_array(second_input_of_tile.value)):
+            return False
+
+        reshape_node = match['reshape']
+        new_shape = reshape_node.in_port(1).data.get_value()
+        if new_shape is None or input_rank_of_unsqueeze != len(new_shape):
+            return False
+
+        return True
+
+    def replace_pattern(self, graph: Graph, match: dict):
+        if not self.is_applicable(match):
             return
 
-        scale = float32_array([second_input_of_tile.value[d_idx]])
+        unsqueeze_node = match['unsqueeze']
+        unsqueeze_name = unsqueeze_node.soft_get('name', unsqueeze_node.id)
+        second_input_of_unsqueeze = unsqueeze_node.in_port(1).get_connection().get_source().node
+        d_idx = int(second_input_of_unsqueeze.value)
         axis = d_idx - 1
-        axis_node = Const(graph, {'name': unsqueeze_name + '/axis', 'value': int64_array([axis])}).create_node()
 
         shape_node = Shape(graph, dict(name=unsqueeze_name + '/Shape')).create_node()
-        scales_node = Const(graph, dict(name=unsqueeze_name + '/scales', value=scale)).create_node()
-        mul_node = Mul(graph, dict(name=unsqueeze_name + '/Mul')).create_node()
-        scales_node.out_port(0).connect(mul_node.in_port(1))
-
-        slice_begin = Const(graph, dict(name=unsqueeze_name + '/slice_begin', value=int64_array([axis]))).create_node()
-        slice_end = Const(graph, dict(name=unsqueeze_name + '/slice_end', value=int64_array([axis + 1]))).create_node()
-
-        strided_slice_node = StridedSlice(graph,
-                                          {'name': unsqueeze_name + '/StridedSlice',
-                                           'begin_mask': int64_array([1]),
-                                           'end_mask': int64_array([1]),
-                                           'new_axis_mask': int64_array([0]),
-                                           'shrink_axis_mask': int64_array([0]),
-                                           'ellipsis_mask': int64_array([0]),
-                                           }).create_node()
+        strided_slice_node = create_op_with_const_inputs(graph,
```

Comment:
What is an idea of this `StridedSlice` because you don't mention it in the comment?

Reply:
An idea is to get `input_shape[axis]`.